### ISSUE TITLE
Can this still be used clientside?

### ISSUE THREAD

**Issue by flatsiedatsie** [vladmandic/face-api#4]
I'm getting complaints from javascript about "require".

There is an example folder, but the example doesn't work because it's missing the basic "face-api.js" in the dist folder? The file `https://vladmandic.github.io/face-api/dist/face-api.js` that the index.html calls doesn't seem to exist anymore?

**Comment by vladmandic** [comment]
Ah, I've switched to ESM build and didn't update the demo, will do.

Require is a NodeJS thing and works with browsers if you run through bundler (webpack, rollup, esbuild, etc). 

You can either use <script> or import with type module.

I'll update the docs Monday.

**Comment by vladmandic** [comment]
I've updated the Face-API library, bumped up TFJS to 2.6.0, updated example and re-wrote docs.
New version is published, let me know how it goes.

**Comment by vladmandic** [comment]
@flatsiedatsie any updated on this issue?

**Comment by flatsiedatsie** [comment]
I will look into it asap.

**Comment by flatsiedatsie** [comment]
The example seems to work clienside now. It loads 6 images and analyses them. It seemed a bit slower than I'm used to the original face-api.js code?

What were your goals with this upgrade to tensorflow 2.0? Would that theoretically make it faster?

Also, I've created a way to run the original Face-Api.js as [a web worker](https://www.hownormalami.eu). Do you think that would stil be possible with this code?

In other words: what are the main things that have changed, and why?

**Comment by vladmandic** [comment]
Why? In short - because I needed it - I'm using TFJS for quite a few different things inside the same project and Face-API being dependent on 1.x meant I couldn't upgrade TFJS in my project as it was causing version conflicts and I needed access to new functions.

Unfortunately, port to TFJS 2.0+ was much deeper than I'd expected as two versions are quite different, so it ended up being a separate project instead of pull request.

And since FaceAPI was open-source, I wanted to release this port as open-source as well (not to steal credits, but to share the work).

Regarding performance: actual execution should be similar as that has not changed. But initialization in TFJS 2.0 is definitely slower as its not a monolithic library, but a modular one which dynamically registers backend functions it needs.

On the other hand, TFJS 2.0 has far more accelerated functions available in newer backends. Doesn't make a difference for FaceAPI since it doesn't use them, but critical for newer models (EfficientNet, CenterNet, etc.).

**Comment by vladmandic** [comment]
regarding web-worker - yes, your code *should* work with this version.  
i've updated the docs on what's different in this version.

**Comment by flatsiedatsie** [comment]
> But initialization in TFJS 2.0 is definitely slower as its not a monolithic library, but a modular one which dynamically registers backend functions it needs.

That's odd. You'd expect something that only loads things it needs to be faster?

Thanks for the explanation. I'm very curious what you're working on now...

**Comment by vladmandic** [comment]
@flatsiedatsie short story, it's not just modularization (that does make it smaller as you can load only parts you need(, it's the dynamic registration of functions in the backends. e.g, what does `tf.mul()` or `tf.add() or `tf.div()`` mean? it used to be a static map towards each implemented version (`cpu`, `webgl`). now, backends perform tests and register functions they implement upon loading.
so instead of static list of functions, they can now be extended on-the-fly. but that costs time during initialization.

btw, if you need fast initialization, you can try `wasm` backend - execution is not as fast as `webgl`, but unlike `cpu` backend, its good enough for simple usage and it starts nearly instantly (must faster startup than `webgl`). although you do need a relatively new browser that supports `wasm` to start with.

what i'm working on? i've started on my library - [`@vladmandic/human`](https://github.com/vladmandic/human)
i love `face-api`, but its showing its age.

**Comment by ost12666** [comment]
maybe the webgl taking a lot of time is time it takes to compile? what we did is warming it up by detecting a dummy blank image and maybe this can also be done in web worker so it does not block CPU and it just means stuff will start to work later which is OK by our use case

**Comment by vladmandic** [comment]
makes sense.
i do work with web workers in my new library as well.

**Comment by flatsiedatsie** [comment]
Whoa. Human. 0_0
